Source/multi.cpp

```diff
@@ -433,7 +433,7 @@ static void multi_process_tmsgs()
 	int cnt;
 	TPkt pkt;
 
-	while (cnt = tmsg_get((BYTE *)&pkt, 512)) {
+	while ((cnt = tmsg_get((BYTE *)&pkt, 512))) {

```

Comment:
The extra parenthesis make no sense to me here. Any specific reason why we are mandating them?

Reply:
Extra parenthesis are the conventional way to indicate that the assignment is intentional, i.e. `=` is not a typo of `==`

Reply:
Are you not familiar with this convention? I remember it existing 20 years ago and it's part of every codebase I've ever worked on and is built into C and C++ linters.

I'm not sure how it came to be but it's rather well-established

Reply:
An alternative to this that will read more clear to those unfamiliar with the convention is to make comparison explicit:

```c++
while ((cnt = tmsg_get((BYTE *)&pkt, 512)) != 0) {
```

Reply:
When the comparison is explicit, the extra parentheses are required, which may be where this convention came from.

I personally prefer explicit comparisons in C++ code but this is a rather controversial opinion (especially in C code).

Reply:
Actually that's what our code style states so I should just apply this.